var/spack/repos/builtin/packages/sundials/package.py

```diff
@@ -832,31 +832,35 @@ def build_smoke_tests(self):
                     self.run_test(exe=cmake_bin, options=["."])
                 self.run_test(exe="make")
 
-    def run_smoke_tests(self):
-        if "~examples-install" in self.spec:
-            return
+    def test_smoke(self):
+        """Run sundials smoke tests"""
 
-        for smoke_test in self._smoke_tests:
-            self.run_test(
-                exe=join_path(self._smoke_tests_path, smoke_test[0]),
-                options=smoke_test[1],
-                status=[0],
-                installed=True,
-                skip_missing=True,
-                purpose=smoke_test[2],
-            )
+        cmake_bin = self.cmake_bin(set=False)
+
+        if not cmake_bin:
+            raise SkipTest("cmake_bin_path.txt not found")
```

Comment:
This is the `cmake` hack that is supposed to be removed along with lines 795-810.  It follows the same pattern as the other PRs that used it.